drivers/gyro/adxrs290/adxrs290.h

```diff
@@ -0,0 +1,223 @@
+#define ADXRS290_REG_SN2			0x06
+#define ADXRS290_REG_SN3			0x07
+#define ADXRS290_REG_DATAX0			0x08
+#define ADXRS290_REG_DATAX1			0x09
+#define ADXRS290_REG_DATAY0			0x0A
+#define ADXRS290_REG_DATAY1			0x0B
+#define ADXRS290_REG_TEMP0			0x0C
+#define ADXRS290_REG_TEMP1			0x0D
+
+#define ADXRS290_REG_POWER_CTL			0x10
+#define ADXRS290_REG_FILTER			0x11
+#define ADXRS290_REG_DATA_READY			0x12
+
+#define ADXRS290_READ				BIT(7)
+#define ADXRS290_TSM				BIT(0)
+#define ADXRS290_MEASUREMENT			BIT(1)
+#define ADXRS290_DATA_RDY_OUT			BIT(0)
+#define ADXRS290_SYNC_MASK			0x03
+#define ADXRS290_SYNC(x)			(x) & ADXRS290_SYNC_MASK
+#define ADXRS290_LPF_MASK			0x07
+#define ADXRS290_LPF(x)				(x) & ADXRS290_LPF_MASK
+#define ADXRS290_HPF_MASK			0xF0
+#define ADXRS290_HPF(x)				((x) & ADXRS290_HPF_MASK )>>4
+
+#define ADXRS290_READ_REG(reg)	(ADXRS290_READ | (reg))
+
+#define ADXRS290_MAX_TRANSITION_TIME_MS 100
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+/**
+ * @enum adxrs290_mode
+ * @brief Mode of the adxrs290
+ */
+enum adxrs290_mode {
+	/** Standby mode */
+	ADXRS290_MODE_STANDBY,
+	/** Measurement mode */
+	ADXRS290_MODE_MEASUREMENT,
+};
+
+/**
+* @enum adxrs290_channel
+* @brief Channel of teh adxrs290 data rate
+*/
+enum adxrs290_channel {
+	/** X-Axis */
+	ADXRS290_CHANNEL_X,
+	/** Y-Axis */
+	ADXRS290_CHANNEL_Y,
+	/** Temp */
+	ADXRS290_CHANNEL_TEMP,
+};
+
+/**
+ * @enum adxrs290_lpf
+ * @brief Low-Pass filter pole location
+ */
+enum adxrs290_lpf {
+	ADXRS290_LPF_480HZ,
+	ADXRS290_LPF_320HZ,
+	ADXRS290_LPF_160HZ,
+	ADXRS290_LPF_80HZ,
+	ADXRS290_LPF_56HZ6,
+	ADXRS290_LPF_40HZ,
+	ADXRS290_LPF_28HZ3,
+	ADXRS290_LPF_20HZ
+};
+
+/**
+ * @enum adxrs290_hpf
+ * @brief High-Pass filter pole location
+ */
+enum adxrs290_hpf {
+	ADXRS290_HPF_ALL_PASS,
+	ADXRS290_HPF_0HZ011,
+	ADXRS290_HPF_0HZ022,
+	ADXRS290_HPF_0HZ044,
+	ADXRS290_HPF_0HZ087,
+	ADXRS290_HPF_0HZ175,
+	ADXRS290_HPF_0HZ350,
+	ADXRS290_HPF_0HZ700,
+	ADXRS290_HPF_1HZ400,
+	ADXRS290_HPF_2HZ800,
+	ADXRS290_HPF_11HZ30
+};
+
+/**
+ * @struct adxrs290_init_param
+ * @brief Device driver initialization structure
+ */
+struct adxrs290_init_param {
+	/** SPI Initialization structure. */
+	struct spi_init_param	spi_init;
+	/** Initial Mode */
+	enum adxrs290_mode mode;
+	/** Initial lpf settings */
+	enum adxrs290_lpf lpf;
+	/** Initial hpf settings */
+	enum adxrs290_hpf hpf;
+
+};
+
+/**
+ * @struct adxrs290_dev
+ * @brief Device driver handler
+ */
+struct adxrs290_dev {
+	/** SPI handler */
+	struct spi_desc *spi_desc;
+};
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+
+/** Read device register. */
```

Comment:
Please use /* */ with one *.
With ** is for the doxygen format and this function are documented already in the .c file. It may cause some errors when building the documentation.